src/formats/apple-notes/convert-note.ts

```diff
@@ -0,0 +1,385 @@
+		
+	async format(table = false): Promise<string> {
+		let fragments = Array.from(this.parseTokens());
+		let firstLineSkip = !table && this.importer.omitFirstLine;
+		let converted = '';
+		
+		for (let j = 0; j < fragments.length; j++) {		
+			let {attr, fragment} = fragments[j];
+			
+			if (firstLineSkip) {
+				if (fragment.contains('\n') || attr.attachmentInfo) firstLineSkip = false;
+				else continue;
+			}
+
+			attr.fragment = fragment;		
+			attr.atLineStart = fragments[j - 1]?.fragment.contains('\n') || j == 0;
+			
+			converted += this.formatMultiRun(attr);
+			
+			if (!/\S/.test(attr.fragment) || this.multiRun == ANMultiRun.Monospaced) {
+				converted += attr.fragment;
+			}
+			else if (attr.attachmentInfo) {
+				converted += await this.formatAttachment(attr);
+			}
+			else if (attr.superscript || attr.underlined || attr.color || attr.font || this.multiRun == ANMultiRun.Alignment) {
+				//non-html stuff doesn't play nicely with html formatting in live preview so we go all html
+				converted += this.formatHtmlAttr(attr);
+			}
+			else {
+				converted += this.formatAttr(attr);
+			}
+		}
+		
+		if (this.multiRun != ANMultiRun.None) converted += this.formatMultiRun(null);
+		if (table) converted.replace('\n', '<br>').replace('|', '&#124;');
+		
+		return converted;
+	}
+	
+	/** Format things that cover multiple ANAttributeRuns. */
+	formatMultiRun(attr: ANAttributeRun | null): string {
+		const styleType = attr?.paragraphStyle?.styleType;
+		let prefix = '';
+		
+		switch (this.multiRun) {
+			case ANMultiRun.List:
+				if (
+					(attr?.paragraphStyle?.indentAmount == 0 && 
+					!LIST_STYLES.includes(styleType!)) ||
+					isBlockAttachment(attr!)
+				) {
+					this.multiRun = ANMultiRun.None;
+				}
+				break;
+			
+			case ANMultiRun.Monospaced:
+				if (styleType != ANStyleType.Monospaced || !attr) {
+					this.multiRun = ANMultiRun.None;
+					prefix += '```\n';
+				}
+				break;
+			
+			case ANMultiRun.Alignment:
+				if (!attr?.paragraphStyle?.alignment || !attr) {
+					this.multiRun = ANMultiRun.None;
+					prefix += '</p>\n';
+				}
+				break;
+		}
+		
+		//separate since one may end and another start immediately
+		if (this.multiRun == ANMultiRun.None) {
+			if (styleType == ANStyleType.Monospaced) {
+				this.multiRun = ANMultiRun.Monospaced;
+				prefix += '\n```\n';
+			}
+			else if (LIST_STYLES.includes(styleType!)) {
+				this.multiRun = ANMultiRun.List;
+				
+				//Apple Notes lets users start a list as indented, so add a initial non-indented bit to those
+				if (attr?.paragraphStyle?.indentAmount) prefix += '\n- &nbsp;\n';
+			}
+			else if (attr?.paragraphStyle?.alignment) {
+				this.multiRun = ANMultiRun.Alignment;
+				const val = this.convertAlign(attr?.paragraphStyle?.alignment);
+				prefix += `\n<p style="text-align:${val};margin:0">`;
+			}
+		}
+		
+		return prefix;
+	}
+	
+	formatHtmlAttr(attr: ANAttributeRun): string {
+		if (attr.strikethrough) attr.fragment = `<s>${attr.fragment}</s>`;
```

Comment:
In Live Preview, doing something like `a<sub>~~b~~</sub>` doesn't work (at least on my end). So if there are things that can only be done in HTML tags what's inside them is done in HTML tags too.

Reply:
Ofc it's your guys' call, but I would recommend against this. A bunch of my CS lecture notes that I used to test used subscript, superscript etc. for formatting; if that was stripped, it could change the meaning of things without my realising if I were to look at them later.

Outside the specialised cases like this, the vast majority of formatting for the notes I used to test ended up being in Markdown, so it's not like it is obtrusive. If it led to the notes being littered with HTML tags I wouldn't have done it.

Reply:
I'm ok leaving it like this for now. It doesn't impact users who are not using super/subscripts/undeliner/color/etc.